disnake/interactions/application_command.py

```diff
@@ -402,7 +402,7 @@ def __init__(
             factory, ch_type = _threaded_channel_factory(channel["type"])
             if factory:
                 channel["position"] = 0  # type: ignore
-                self.channels[int(str_id)] = factory(guild=guild, state=state, data=channel)  # type: ignore
+                self.channels[int(str_id)] = guild.get_channel(int(str_id)) or factory(guild=guild, state=state, data=channel)  # type: ignore
```

Comment:
`guild` might be `None` here? Not sure if you even get a channel object outside of guilds, may need to test that